### PR TITLE
[Auto Downloading] Update UI for Global Auto Downloading

### LINKED ISSUE #2970: [Project] Auto Downloading Episodes - Implement Global Screen settings
Figma: VvERCFIopLjFL8Fnczsf8r-fi-1_10

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- This PR updates the UI for auto downloading settings (global) by adding the limit setting
- Adds local feature flag to wrap this change
- Add tracks for when the user selects a limit downloading
- Context: pdeCcb-744-p2
- Figma: VvERCFIopLjFL8Fnczsf8r-fi-1_756

Fixes #2970

### AUTO_DOWNLOAD feature flag OFF
1. Have the FF disabled
2. Go to Profile -> Settings -> Auto Download
3. ✅ Ensure that New Episodes toggle does not have description
4. ✅ Ensure you don't see the limit download section

### AUTO_DOWNLOAD feature flag ON
1. Have the FF enabled
2. Go to Profile -> Settings -> Auto Download
3. ✅ Ensure that New Episodes toggle has description
4. ✅ Ensure you see the limit download section
5. Tap on limit downloads label and select one option
6. ✅ Ensure your option is selected
7. ✅ Ensure 🔵 Tracked: settings_auto_download_limit_downloads_toggled, Properties: {"value":"value"... is tracked

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>AutoDownloadLimitSetting</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by CookieyedCodes** [comment]
Question is this limit total downloads or a per podcast setting, I personally would prefer a limit to the number of episodes downloaded from the top of up next rather then a global option 🤔

File: modules/services/localization/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1145,6 +1145,16 @@
     <string name="settings_auto_download_filters_episodes">All filter episodes</string>
     <string name="settings_auto_download_new_episodes">New episodes</string>
     <string name="settings_auto_download_podcasts">Auto download podcasts</string>
+    <string name="settings_auto_download_new_episodes_description">Download new episodes as they’re released, and manage your storage by setting a limit on how many episodes are saved</string>
```
Comment (geekygecko) [inline modules/services/localization/src/main/res/values/strings.xml:1148]:
```suggestion
    <string name="settings_auto_download_new_episodes_description">Download new episodes as they’re released, and manage your storage by setting a limit on how many episodes are saved.</string>
```
Other option descriptions have a full stop, so maybe we should match this.

Screenshot_20241008_211053

**Review by geekygecko — APPROVED** [review]
Looks good. I adding some minor comments.

File: modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt [hunk 2]
```diff
@@ -385,6 +386,14 @@ class SettingsImpl @Inject constructor(
         toString = { it.preferenceInt.toString() },
     )
 
+    override val autoDownloadLimit = UserSetting.PrefFromString(
+        sharedPrefKey = "autoDownloadLimit",
+        defaultValue = AutoDownloadLimitSetting.TWO_LATEST_EPISODE,
+        sharedPrefs = sharedPreferences,
+        fromString = { AutoDownloadLimitSetting.fromPreferenceString(it) },
```
[the remark this answers is not in the document]
Comment (mebarbosa) [reply modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt:393]: Makes sense! I was following the existing example of `PlayOverNotificationSetting`. I will update this

**Pull request merged**